datasetinsights/estimators/faster_rcnn.py

```diff
@@ -22,6 +22,594 @@
 MAX_BOXES_PER_IMAGE = 100
 logger = logging.getLogger(__name__)
 DEFAULT_ACCUMULATION_STEPS = 1
+TRAIN = "train"
+VAL = "val"
+TEST = "test"
+
+
+class FasterRCNN(Estimator):
+    """
+    https://github.com/pytorch/vision/tree/master/references/detection
+    https://arxiv.org/abs/1506.01497
+    Args:
+        config (CfgNode): estimator config
+        writer: Tensorboard writer object
+        checkpointer: Model checkpointer callback to save models
+        device: model training on device (cpu|cuda)
+        local_rank (int): (optional) rank of process executing code
+        gpu (int): (optional) gpu id on which code will execute
+    Attributes:
+        model: pytorch model
+        writer: Tensorboard writer object
+        checkpointer: Model checkpointer callback to save models
+        device: model training on device (cpu|cuda)
+    """
+
+    def __init__(
+        self,
+        *,
+        config,
+        writer,
+        kfp_writer,
+        checkpointer,
+        device,
+        box_score_thresh=0.05,
+        gpu=0,
+        rank=0,
+        **kwargs,
+    ):
+        logger.info(f"initializing faster rcnn")
+        self.config = config
+        self.device = device
+        self.writer = writer
+        self.kfp_writer = kfp_writer
+        model_name = f"fasterrcnn_{self.config.backbone}_fpn"
+        self.model = torchvision.models.detection.__dict__[model_name](
+            num_classes=config.num_classes,
+            pretrained_backbone=config.pretrained_backbone,
+            pretrained=config.pretrained,
+            box_detections_per_img=MAX_BOXES_PER_IMAGE,
+            box_score_thresh=box_score_thresh,
+        )
+        self.model_without_ddp = self.model
+        self.gpu = gpu
+        self.rank = rank
+        self.sync_metrics = config.get("synchronize_metrics", True)
+        logger.info(f"gpu: {self.gpu}, rank: {self.rank}")
+
+        self.checkpointer = checkpointer
+        checkpoint_file = config.get("checkpoint_file", const.NULL_STRING)
+        if checkpoint_file != const.NULL_STRING:
+            checkpointer.load(self, config.checkpoint_file)
+
+        self.metrics = {}
+        for metric_key, metric in config.metrics.items():
+            self.metrics[metric_key] = EvaluationMetric.create(
+                metric.name, **metric.args
+            )
+        self.model.to(self.device)
+
+        if self.config.system.distributed:
+            self.model = torch.nn.parallel.DistributedDataParallel(
+                self.model, device_ids=[self.gpu]
+            )
+            self.model_without_ddp = self.model.module
+
+    def train(self, **kwargs):
```

Comment:
Docstring? Whant's the input parameter here?

Reply:
No input parameter, its simple method call on object.